components/fighter.py

```diff
@@ -119,43 +123,41 @@ def heal(self, amount: int) -> int:
 
         return amount_recovered
 
-    def take_damage(self, incoming_dmg: int, damage_type: str=None) -> None:
-        modified_dmg = incoming_dmg
+    def decrease_hp(self, amount: int) -> int:
+        """ Deal damage to the Fighter
+
+        Damages the Fighter and returns the amount of damage dealt which
+        currently is always the same as the incoming damage.
+        """
+
+        self.hp -= amount
+        return amount
+
+    def damage(self, incoming_dmg: int, damage_type: str=None) -> str:
+        """ Attempt to damage the Fighter
+
+        This will check for resistances, etc. and may result in a heal.
+        It will return string indicating the outcome.
+        """
         if damage_type:
```

Comment:
This will FALSE on 0.0

Reply:
one option: is not None
another option: default the dmg to 1.0 when nothing is specified